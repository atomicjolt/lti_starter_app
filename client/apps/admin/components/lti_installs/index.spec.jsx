import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import TestRenderer from 'react-test-renderer';
import Index from './index';

const mockStore = configureStore([]);
const applicationId = '123';
const applicationInstanceId = '4847';

const store = mockStore({
  settings: {
    sign_out_url: 'https://www.example.com',
  },
  applicationInstances: {
    applicationInstances: [],
  },
  accounts: {
    accounts: {
      1234: {
        id: 1234,
        parent_account_id: null,
        name: 'paul',
      },
      2345: {
        id: 2345,
        parent_account_id: 1234,
        name: 'joe',
      },
    }
  },
  applications: {},
  courses: [{}],
  applicationInstance: {},
  loadingCourses: {},
  loadingAccounts: false,
  sites: {},
});

describe('the index component', () => {
  let result;
  let instance;

  beforeEach(() => {
    result = TestRenderer.create(
      <Provider store={store}>
        <Index params={{
          applicationId,
          applicationInstanceId,
        }}
        />
      </Provider>
    );
    instance = result.root;
  });
  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('sets the active account', () => {
    const buttons = instance.findAllByType('button');
    const button = buttons[1];
    const name = button.props.children[1];
    expect(name).toEqual('paul');
  });
});
