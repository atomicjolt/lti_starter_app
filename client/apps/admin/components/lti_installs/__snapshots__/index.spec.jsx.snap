// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the index component matches the snapshot 1`] = `
<div
  style={
    Object {
      "height": "100%",
    }
  }
>
  <Connect(Heading) />
  <div
    className="o-contain"
  >
    <Sidebar
      accounts={Object {}}
      applicationInstance={Object {}}
      canvasRequest={[Function]}
      currentAccount={null}
      saveApplicationInstance={[Function]}
      setAccountActive={[Function]}
      sites={Object {}}
    />
    <Connect(InstallPane)
      account={null}
      applicationInstance={Object {}}
      canvasRequest={[Function]}
      loadExternalTools={[Function]}
      loadingCourses={Object {}}
    />
  </div>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="c-modal c-modal--site is-open c-modal--error loading"
    closeTimeoutMS={0}
    contentElement={[Function]}
    contentLabel="Modal"
    isOpen={false}
    overlayClassName="c-modal__background"
    overlayElement={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    preventScroll={false}
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <div
      className="c-loading-icon"
    />
       Loading...
  </Modal>
</div>
`;
